src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,1142 @@
+        String outputFilePath = new File(OUTPUT_FILE_NAME).getCanonicalPath();
+
+        ForDevelopersApi forDevelopersApi =  developer.getClient(ForDevelopersApi.class);
+        CmsPublicKey publicKey = forDevelopersApi.getStudyPublicCsmKey().execute().body();
+
+        StudyUploadEncryptor.writeTo(publicKey.getPublicKey(), inputFilePath, outputFilePath);
+    }
+
+    private boolean isProduction(Environment env) {
+        return env == Environment.PROD;
+    }
+
+    private void createSynapseProjectAndTeam() throws IOException, SynapseException {
+        StudiesApi studiesApi = developer.getClient(StudiesApi.class);
+        Study currentStudy = studiesApi.getUsersStudy().execute().body();
+        currentStudy.setSynapseDataAccessTeamId(null);
+        currentStudy.setSynapseProjectId(null);
+        currentStudy.setDisableExport(false); // make this study exportable
+
+        studiesApi.updateUsersStudy(currentStudy).execute().body();
+
+        // execute
+        studiesApi.createSynapseProjectTeam(ImmutableList.of(testUserId.toString())).execute().body();
+
+        // save project and team id
+        Study newStudy = studiesApi.getUsersStudy().execute().body();
+        assertEquals(newStudy.getIdentifier(), currentStudy.getIdentifier());
+        String projectId = newStudy.getSynapseProjectId();
+        Long teamId = newStudy.getSynapseDataAccessTeamId();
+
+        Entity project = synapseClient.getEntityById(projectId);
+        assertNotNull(project);
+        assertEquals(project.getEntityType(), "org.sagebionetworks.repo.model.Project");
+        this.project = (Project) project;
+        Team team = synapseClient.getTeam(teamId.toString());
+        assertNotNull(team);
+        this.team = team;
+    }
+
+    private static String testUpload(String filePath) throws Exception {
+        // set up request
+        File file = new File(filePath);
+
+        ForConsentedUsersApi usersApi = user.getClient(ForConsentedUsersApi.class);
+        UploadSession session = RestUtils.upload(usersApi, file);
+
+        String uploadId = session.getId();
+
+        // get validation status
+        UploadValidationStatus status = null;
+        for (int i = 0; i < UPLOAD_STATUS_DELAY_RETRIES; i++) {
+            Thread.sleep(UPLOAD_STATUS_DELAY_MILLISECONDS);
+
+            status = usersApi.getUploadStatus(session.getId()).execute().body();
+            if (status.getStatus() == UploadStatus.VALIDATION_FAILED) {
+                // Short-circuit. Validation failed. No need to retry.
+                fail("Upload validation failed, UploadId=" + uploadId);
+            } else if (status.getStatus() == UploadStatus.SUCCEEDED) {
+                break;
+            }
+        }
+        // userClient.upload marks the download complete
+        // marking an already completed download as complete again should succeed (and be a no-op)
+        user.getClient(ForConsentedUsersApi.class).completeUploadSession(session.getId());
```

Comment:
RestUtils.upload() already calls Complete Upload. Which is how the upload succeeds validation in the first place.